### ISSUE TITLE
Broken Scenes

### ISSUE THREAD

**Issue by teufelweich** [apple/ml-hypersim#22]
I've noticed that there are scenes with pixels belonging to the class person. I took a closer look to theses scenes and discovered some broken renderings.

`ai_004_009/images/scene_cam_01_final_preview` contains completely broken images. Seems like there is a light missing in that scene. But even the textures don't make any sense. This scene confuses even me, as a human.

`frame.0001.diffuse_reflectance.jpg`
![frame.0001.diffuse_reflectance.jpg](https://user-images.githubusercontent.com/33784413/117446429-5a21df80-af3c-11eb-96fe-a7e5af2ae7c3.png)

`frame.0001.tonemap.jpg`
![frame.0001.tonemap.jpg](https://user-images.githubusercontent.com/33784413/117446802-eb915180-af3c-11eb-8692-40def1cbbef4.png)

`ai_038_007/images/scene_cam_00_final_preview` is missing textures completely. The floating rectangles are labeled as persons.

`frame.0014.tonemap.jpg`
![frame.0014.tonemap.jpg](https://user-images.githubusercontent.com/33784413/117447272-90ac2a00-af3d-11eb-9227-ca3c7a2e10dd.png)

`frame.0014.semantic.png`
![frame.0014.semantic.png](https://user-images.githubusercontent.com/33784413/117447484-e1238780-af3d-11eb-9a63-2ffb408a7fa6.png)

These are the first scenes I looked into. Both of them are included in the train split and are detrimental for training in my opinion.
Are you aware of such issues? How many more broken scenes could there be? 

Despite this, thanks for your work.

**Comment by mikeroberts3000** [comment]
I agree that `ai_038_007` isn't very useful for segmentation tasks, but Hypersim can also be used for other computer vision tasks beyond segmentation.

Unfortunately I don't have a list of scenes affected by this texture rendering issue, but I know there are at most two or three such scenes. You'll be able to find these others by looking at the first image for each scene. There are only 461 scenes, so you won't need to do too much manual inspection.

Those depth distributions look reasonable to me. Your interpretation of the spikes also sounds reasonable. There must be occasional frames where the camera is exactly aligned with a wall. I'm sure this phenomenon occasionally happens in the training set too, but perhaps it's less noticeable in your plots because the training set is much bigger.

I think this phenomenon can arise naturally when an artist-placed camera happens to be axis-aligned, and the walls in the scene also happen to be axis-aligned. And it is always possible (but unlikely) that our randomly selected views could happen to align perfectly with a wall. So this phenomenon is not necessarily indicative of a broken scene or camera trajectory or frame. That being said, it seems plausible to me that a perfectly axis-aligned camera is more likely to have been accidentally spawned by the artist, and less likely to have been carefully placed by the artist to frame the scene in a particular way. Therefore, when a camera is perfectly axis-aligned, perhaps it is a hint that the camera does not convey artistic intent. Of course, this reasoning only holds if the walls in the scene are also axis-aligned, but they usually are.

**Comment by teufelweich** [comment]
My simple search found another pitch black scene in the training split: `(ai_003_001, cam_00)`
In the test and val split I've found none.

Thats true, I completely forgot that there are other tasks apart from the ones I'm working on 😅

That's a good explanation.
Maybe I could search for these monotone depth images with a heuristic like `np.std(depth) < 1m`. A straight faced wall without objects in the frame would likely render it impossible to estimate the depth of it. I should try a histogram of mean and std of every image.

**Comment by mikeroberts3000** [comment]
Good catch! So the pitch black camera trajectories we've found so far are as follows:

```
ai_003_001, cam_00
ai_004_009, cam_01
```

While you were searching for pitch black camera trajectories, I manually inspected the scenes to look for that texture rendering problem. The scenes I found with the texture rendering problem are as follows:

```
ai_015_006
ai_038_007
ai_046_001
ai_046_009
ai_048_004 (all the objects in this scene are assigned a seemingly random bright color)
ai_053_005
```

I'll update these lists if we find any more. Thanks for helping to dig into these issues 😄

**Comment by teufelweich** [comment]
We've found more broken scenes. This time with black segmentation images or depths.
```
        'ai_012_007': '*',  # train (no depth)
        'ai_013_001': '*',  # train (no depth)
        'ai_023_008': 'cam_01',  # train (no labels)
        'ai_026_020': '*',  # train (no labels)
        'ai_023_009': '*',  # train (no labels)
        'ai_038_007': '*',  # train
        'ai_023_004': 'cam_01',  # train (no labels)
        'ai_023_006': '*',  # train (no labels)
        'ai_026_013': '*',  # train (no labels)
        'ai_026_018': '*',  # train (no labels)
        # 'ai_044_004': 'cam_01',  # test (no labels)
```

**Comment by mikeroberts3000** [comment]
Good catch, thank you!
